src/hooks/useQuery.ts

```diff
@@ -33,19 +33,33 @@ export type UseQueryResponse<T> = [
 export const useQuery = <T = any, V = object>(
   args: UseQueryArgs<V>
 ): UseQueryResponse<T> => {
+  const isMounted = useRef(false);
   const unsubscribe = useRef(noop);
+  const initialState = useRef({ fetching: true });
+  const updateState = useRef(update => {
+    initialState.current = update;
+  });
 
   const client = useContext(Context);
+
   const request = useMemo(
     () => createRequest(args.query, args.variables as any),
     [args.query, args.variables]
   );
 
-  const [state, setState] = useState<UseQueryState<T>>({
-    fetching: false,
-    error: undefined,
-    data: undefined,
-  });
+  if (!isMounted.current) {
```

Comment:
I really like the idea of reusing `isMounted` for this 👌 

I suppose we could also reuse `executeQuery = useCallback( /* ... */` now?

Maybe it could check whether `isMounted` is still `false` to update `initialState`? That would also mean that `updateState` becomes unnecessary.

I'm also not sure how `useState` works with Fiber but `setState` on `Component` was always batched, so it would only update state once. This even worked when the `Component` was still mounting. At least during researching the current SSR implementation I found out that `useState` still works similarly to that there. So if it also does in non-SSR then we could simply call `setState` and not worry about `initialState` either, since it might be batched into the first render

Reply:
I don't see how we can reuse executeQuery since to use that we already need  our setState defined, that useState inturns relies on initialState.current.

I did implement the second remove `updateState` part,  will ook more in depth later. Thanks for your excellent and through guidance. Amazing!

Reply:
Yea, true, reusing `executeQuery` definitely hinges on whether `useState` can again come before it and whether it's actually going to change the `initialState`. At least in SSR this works because calling it triggers another "loop" through the component, so not technically a rerender but a rerun. Not sure if that's the case in normal, non-SSR React

Reply:
I've added a few suggestions but if I'm not mistaken, the issue described leaves us only concerned about the initial state on mount.

It looks like we have two scenarios:
 - Pause is false and the initial state is  { fetching: true }
 - Pause is true and the initial state is { fetching: false }

If that's the case, we can make this change without the need for refs or additional logic.

```tsx
const [state, setState] = useState<UseQueryState<T>>({
  fetching: args.pause !== true,
  error: undefined,
  data: undefined,
});
```

Reply:
The other scenario is some preparation for suspense SSR so we'll also would want to keep the executeQuery excursion that runs on initial mount outside of useEffect 

But depending on how React works the executeQuery useCallback could maybe be reused for that, simplifying this PR. We'll see how that pans out I guess. Might or might nor work 😅

Reply:
> SSR

I'm out of here :eyes: 

Are you saying that following SSR, the state and ref values are not synchronized from the backend?

Reply:
Up to you what this PR becomes, my initial solution was to make the initial state based on pause. If the current wip helps out more for future work maybe we can make a seperate simple solution PR and keep this one for the experimental work.